install.sh

```diff
@@ -1,45 +1,135 @@
+COPY='rsync -uv ' # Just because some platforms don't know
+				  # `cp -u`!
+S_PLATFORM=`uname`
+IS_LINUX=
+
+if [ "${S_PLATFORM}" == 'Linux' ]; then
+	COPY='cp -uv '
+	IS_LINUX=1
+fi
+
+$COPY -b ./.env.dist ./.env
+if [ ! -z "${IS_LINUX}" ]; then
+	sed -i -E "s|USER_ID=(.*)|USER_ID=$(id -u)\:$(id -g)|g" ./.env
+fi
 
 # Use configuration for environment vars
 source ./.env
-
-# Ensure we have a folder to work with
-if [ -z "$1" ]; then
-	echo "Please specify Akeneo git folder"
+if [ -z "${PIM_DB_NAME}" ] || [ -z "${PIM_DB_USER}" ]; then
+	echo ""
+	echo "Please set your environment values in '.env' file "
+	echo "and try again. Thank you."
+	echo ""
 	exit 1
 fi
 
-
-# Make the connection
-source ./.connection.sh
+DOCKER_EXEC="docker exec akeneo_pim_app "
+MACHINE_NAME="${MACHINE_NAME-machine}"
 
 function print_msg()
 {
 	printf "$1\n"	
 }
 
 
-function make_process_user()
+## ------------------------------------
+## Installs all project dependencies
+## @param $WEBROOT
+## ---------------------------------------
+function composer_install()
 {
 
-	PIM_WEB_PROCESS_USER= "${PIM_WEB_PROCESS_USER:=$(echo id -u)}"
-	COMMANDS=$(cat <<-EOF
-	if [ \$(id -u alpine > /dev/null 2>&1; echo \$? ) -ne 0 ]; then
-		addgroup -g "${PIM_WEB_PROCESS_USER}" alpine
-    	adduser -D -u "${PIM_WEB_PROCESS_USER}" -G alpine alpine
-	fi; exit
+	
+	printf "\n    About to install vendors. Sit tight ...\n"
+
+	# Check that we can install doctrine/mongodb-odm-bundle
+	if [ $($DOCKER_EXEC composer show  | grep "mongo\-php\-adapter" > /dev/null 2>&1; echo $? ) -ne 0 ]; then
+
+		xcommand="composer require --update-no-dev --no-scripts -v 'doctrine/mongodb-odm-bundle' "
+		 # Check to see if we need to install composer
+		if [ $( $DOCKER_EXEC ${xcommand} 2> /dev/null; echo $?) -ne 0 ]; then
+			# Current Doctrine isn't PHP7 compliant, we'll need to 
+			# Install a middleware to make this backwards compatible
+		 	$DOCKER_EXEC composer require alcaeus/mongo-php-adapter --ignore-platform-reqs --no-scripts 
+		fi # Check that mongodb-odm-bundle can be installed safely
+	else
+		printf "    skipping\n"
+
+	fi # Check for mongo-php-adapter exists
+	
+	# Install the project
+	echo "     Performing PIM seup ..."
+	COMPOSER_INSTALL=$(cat <<-EOF
+	printf "\n\n\n\n\n" | composer install -o
 EOF
 )
-	docker exec akeneo_pim_app /bin/bash -c "${COMMANDS}"
 
+	$DOCKER_EXEC /bin/sh -c "${COMPOSER_INSTALL}"
+
+	printf "\n    Done\n"
+
+}
+
+
+function set_parameter()
+{
+
+CONFIG_COMMAND=$(cat <<-EOF
+
+WEBROOT=/var/www/pim/
+
+if [ -d "\${WEBROOT}" ]; then
+
+	# current php ini file
+	php_ini_file=\$(realpath \$(php --ini | grep -e ".*\Loaded Config" | cut -d':' -f2))
+
+	PIM_TIMEZONE="Australia/Sydney"
+	
+	sed -i -E "s|^;date\.timezone =\\$|date\.timezone = \${PIM_TIMEZONE}|g" "\${php_ini_file}"
+	sed -i -E "s|^date\.timezone =\\$|date\.timezone = \${PIM_TIMEZONE}|g" "\${php_ini_file}"
+
+	# Set the default parameters
+	params_file="\${WEBROOT}/app/config/parameters.yml"
```

Comment:
there's a nice guy called dotenv that solve this issue as all you pass is a environment variable which is supported by almost all system.